lib/membrane_mp4/muxer/cmaf.ex

```diff
@@ -30,6 +33,10 @@ defmodule Membrane.MP4.Muxer.CMAF do
                 type: :time,
                 spec: Membrane.Time.t(),
                 default: 2 |> Time.seconds()
+              ],
+              support_b_frames: [
+                type: :boolean,
+                default: false
```

Comment:
It doesn't seem like something that should be configured by the user. 
1. Making the `false` a default is problematic as we're defaulting to the broken output for b-frame containing streams and that should not be the case.
2. Assuming `true` will be default - I don't really see the point of disabling it manually. I believe we can disable it when profile of the incoming stream specifies it doesn't have B-frames, but enable it otherwise

Reply:
Hmm, as discussed during the meeting, I will leave it as a flag to be set by the user.
Originally, the reason why I have thought about adding such an option was that the tests in the mp4 plugin are based on reference files, which do not use B-frames (and therefore do not have offset field in their samples) - but surely we can update these reference files.

Reply:
Just a note though - it should not be named this way, cause why would you want to disable support for something? (Well the reason is size but you need to know it upfront) What this option does is toggle `sample_offset` generation and I think the name of the option should be derived from that, e.g. `include_sample_offset?` and the description might explain why it is needed.
Additionally, the muxer should raise if this setting is disabled and B-frames are present - that however would require some changes to caps as the H264 profile is not passed in `MP4.Payload` caps

Reply:
Hmm, I have got a point. However, I have decided to update the reference files and get rid of that option - 4B per sample is not that much to be worthwhile :D

Reply:
That works for me :+1: